drivers/serial/uart_xmc4xxx.c

```diff
@@ -66,12 +70,16 @@ static const struct uart_driver_api uart_xmc4xxx_driver_api = {
 };
 
 #define XMC4XXX_INIT(index)						\
+PINCTRL_DT_INST_DEFINE(index);						\
 static struct uart_xmc4xxx_data xmc4xxx_data_##index = {		\
 	.config.baudrate = DT_INST_PROP(index, current_speed)		\
 };									\
 									\
 static const struct uart_xmc4xx_config xmc4xxx_config_##index = {	\
 	.uart = (XMC_USIC_CH_t *)DT_INST_REG_ADDR(index),		\
+	.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(index),			\
+	.input_src = XMC4XXX_PINMUX_GET_ALT(				\
+		DT_INST_PROP_BY_PHANDLE_IDX(index, rx_pin, 0, pinmux)), \
```

Comment:
why is this required?

questions/suggestions:
- could `XMC_UART_CH_SetInputSource` be handled by pinctrl? (you can store periph reg to perform per peripheral actions)
- If you need pin info (e.g. as it happens for i2c bus recovery in some drivers), use `xxx-gpios` and keep it independent of pinctrl

Reply:
> why is this required?

The peripherals on the xmc4500 also use a mux to connect to different gpio pins. I'll think about the two ways you proposed.

> questions/suggestions:
> 
>     * could `XMC_UART_CH_SetInputSource` be handled by pinctrl? (you can store periph reg to perform per peripheral actions)

Yes, but it didn't quite feel right to have peripheral specific code in the pinctrl driver. It would also need to handle different types of peripherals (USIC, CAN, Ethernet). Are there any other pinctrl drivers that do this?

Reply:
> * If you need pin info (e.g. as it happens for i2c bus recovery in some drivers), use `xxx-gpios` and keep it independent of pinctrl

I added a property input-src in a separate commit. Let me know if this is ok.